test/reader_test.go

```diff
@@ -0,0 +1,556 @@
+				// Clear database to ensure idempotency.
+				m.Database("jaeger-tracing").Collection("traces9").Drop(ctx)
+				reader := jaeger_mongodb.NewSpanReader(m.Database("jaeger-tracing").Collection("traces9"), nil)
+				writer := jaeger_mongodb.NewSpanWriter(m.Database("jaeger-tracing").Collection("traces9"), nil)
+				GenerateTraces(ctx, writer, 50, "single", false)
+				traces, err := reader.FindTraces(ctx, &spanstore.TraceQueryParameters{
+					StartTimeMin: time.Date(1997, 04, 30, 05, 1, 1, 1, time.UTC),
+					StartTimeMax: time.Now(),
+					NumTraces:    1500,
+				})
+				if err != nil {
+					log.Fatal(err)
+				}
+				// Ensure we have 50 traces.
+				assert.Equal(t, 50, len(traces))
+				// Ensure all the traces have only 1 span.
+				for _, trace := range traces {
+					assert.Equal(t, 1, len(trace.GetSpans()))
+				}
+			},
+		},
+		{
+			name:     "Test Find Traces -- Tag Filtering",
+			endTs:    time.Date(2030, 9, 2, 7, 30, 15, 100, time.UTC),
+			lookback: twentyYears,
+			runAssertion: func(endTs time.Time, lookback time.Duration) {
+				// Clear database to ensure idempotency.
+				m.Database("jaeger-tracing").Collection("traces10").Drop(ctx)
+				reader := jaeger_mongodb.NewSpanReader(m.Database("jaeger-tracing").Collection("traces10"), nil)
+				writer := jaeger_mongodb.NewSpanWriter(m.Database("jaeger-tracing").Collection("traces10"), nil)
+				GenerateTraces(ctx, writer, 50, "circular", false)
+				traces200, err := reader.FindTraces(ctx, &spanstore.TraceQueryParameters{
+					StartTimeMin: time.Date(1997, 04, 30, 05, 1, 1, 1, time.UTC),
+					StartTimeMax: time.Now(),
+					NumTraces:    1500,
+					Tags: map[string]string{
+						"http.status_code": "200",
+					},
+				})
+				if err != nil {
+					log.Fatal(err)
+				}
+				// Assert there are 45 traces with tag -- http.status_code = 200
+				assert.Equal(t, 45, len(traces200))
+				traces404, err := reader.FindTraces(ctx, &spanstore.TraceQueryParameters{
+					StartTimeMin: time.Date(1997, 04, 30, 05, 1, 1, 1, time.UTC),
+					StartTimeMax: time.Now(),
+					NumTraces:    1500,
+					Tags: map[string]string{
+						"http.status_code": "404",
+					},
+				})
+				if err != nil {
+					log.Fatal(err)
+				}
+				// Assert there are only 5 traces with tag -- http.status_code = 404
+				assert.Equal(t, 5, len(traces404))
+			},
+		},
+	}
+	for _, tc := range testCases {
+		println(tc.name)
+		tc.runAssertion(tc.endTs, tc.lookback)
+		println("====")
+	}
+}
+func BenchmarkTagFiltering(b *testing.B) {
+	m, err := mongo.Connect(context.TODO(), options.Client().
+		ApplyURI(fmt.Sprintf("mongodb://%s:27017", GetLocalIP())).
+		SetWriteConcern(writeconcern.New(writeconcern.W(1))))
+	if err != nil {
+		log.Fatal(err)
+	}
+	reader := jaeger_mongodb.NewSpanReader(m.Database("jaeger-tracing").Collection("traces"), nil)
+
+	ctx := context.TODO()
+	fmt.Println(reader.GetOperations(ctx, spanstore.OperationQueryParameters{}))
```

Comment:
nit; i don't think you need to print these out anymore?

Reply:
removed